### PR TITLE
Add example for checkout

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
🆒

File: base/libgit2/libgit2.jl [hunk 1]
```diff
@@ -458,6 +458,23 @@ Equivalent to `git checkout [-f] --detach <commit>`.
 Checkout the git commit `commit` (a [`GitHash`](@ref) in string form)
 in `repo`. If `force` is `true`, force the checkout and discard any
 current changes. Note that this detaches the current HEAD.
+
+# Example
+
+```julia
+repo = LibGit2.init(repo_path)
+open(joinpath(LibGit2.path(repo),"file1"),"w") do f
```
Comment (Sacha0) [inline base/libgit2/libgit2.jl]: For consistency, perhaps inject spaces between function arguments in this line and the similar `open` call below?

**Review by Sacha0 — APPROVED** [review]
lgtm! :) (Absent objections, requests for time, or someone beating me to it, I plan to merge this ~~tomorrow morning~~ following whitespace fixups. Best!)

**Comment by kshyatt** [comment]
Whitespace addressed, going to merge <3

**Pull request merged**